WordPress/gutenberg
# Update site editor block placeholder styling

Part of #35501.

Unifies the styling of site editor block placeholder states.

| Before | After | 
| - | - |
| <img width="325" src="https://user-images.githubusercontent.com/846565/136789558-9ddb63e0-86b3-4226-b915-550235bbfa13.png"> | <img width="325" alt="Screenshot 2021-10-11 at 13 21 07" src="https://user-images.githubusercontent.com/846565/136789584-c6089309-0413-49f5-b3cd-2f667ca004b0.png"> |

To test:

* Open Site Editor
* Insert the following blocks, ensuring none of them have data references (the 404 template is a good place to test):
  * Post Title
  * Post Content
  * Post Date
  * Post Excerpt
  * Post Author
  * Term Description
  * Next Post
  * Previous Post
  * Archive Title
  * Post Categories
  * Post Tags
 * Ensure placeholder styles like dashed borders, backgrounds, custom margin/padding, etc are removed, and type styling is consistent across blocks.

Note: the opacity applied to the Next/Previous post blocks seems to come from the `RichText` component which I didn't want to touch here.
